DICOMPlugins/DICOMM3DPlugin.py

```diff
@@ -0,0 +1,174 @@
+        loadables.append(loadable)
+
+        logging.debug('DICOM M3D modality found')
+
+    return loadables
+
+  def referencedSeriesName(self,loadable):
+    """Returns the default series name for the given loadable"""
+    referencedName = "Unnamed Reference"
+    if hasattr(loadable, "referencedSeriesUID"):
+      referencedName = self.defaultSeriesNodeName(loadable.referencedSeriesUID)
+    return referencedName
+
+  def GetEncapsulatedDocument(self, loadable):
+    return loadable.EncapsulatedDocument
+  
+  def GetEncapsulatedDocumentLength(self, loadable):
+    return loadable.EncapsulatedDocumentLength
+
+
+  def load(self,loadable):
+    """ Load the DICOM M3D object
+    """
+    logging.debug('DICOM M3D load()')
+    try:
+      uid = loadable.uid
+      logging.debug('in load(): uid = '+uid)
+    except AttributeError:
+      return False
+
+    self.tempDir = os.path.join(slicer.app.temporaryPath, 
+                                "QIICR", "M3D",
+                                  self.currentDateTime, loadable.uid)
+    print(self.tempDir)
+    try:
+        os.makedirs(self.tempDir)
+    except OSError:
+      pass
+    
+    # produces output label map files, one per segment, and information files with
+    # the terminology information for each segment
+    STLFileName = slicer.dicomDatabase.fileForInstance(uid)
+    if STLFileName is None:
+      logging.error('Failed to get the filename from the DICOM database for ' + uid)
+      self.cleanup()
+      return False
+    
+    read_buffer = io.BytesIO(self.GetEncapsulatedDocument(loadable))#bytes(loadable.EncapsulatedDocument))
+    buffer_view = read_buffer.getbuffer()
+    if (int(self.GetEncapsulatedDocumentLength(loadable)) % 2) != 0:
+      buffer_view = buffer_view[0:(read_buffer.getbuffer().nbytes -1)]
+    else:
+      pass
+    
+    STLFilePath = os.path.join(self.tempDir, "temp.STL")
+    
+    with open(STLFilePath, 'wb') as file: 
+      shutil.copyfileobj(read_buffer, file)
+    assert os.path.exists(STLFilePath)
+    file.close()
+    #validity checks !!!!!
+    ####
+    self._createModelNode(loadable, STLFilePath)
+    
+    self.cleanup()
+    
+    return True
+
+  def _createModelNode(self, loadable, STLFilePath):
+    ModelNode = slicer.util.loadModel(STLFilePath)#slicer.mrmlScene.AddNewNodeByClass("vtkMRMLModelNode").AddModel(STLFileName)
+    ModelNode.SetName(loadable.name)
+   
+    ModelDisplayNode = slicer.mrmlScene.AddNewNodeByClass("vtkMRMLModelDisplayNode")
+    ModelNode.SetAndObserveDisplayNodeID(ModelDisplayNode.GetID())
+    self.addSeriesInSubjectHierarchy(loadable, ModelNode)
+    # return ModelNode
+  
+class DICOMM3DPlugin:
+  """
+  This class is the 'hook' for slicer to detect and recognize the plugin
+  as a loadable scripted module
+  """
+  def __init__(self, parent):
+    parent.title = "DICOM M3D Object Import Plugin"
+    parent.categories = ["Developer Tools.DICOM Plugins"]
+    parent.contributors = ["Andrey Fedorov, BWH"]
```

Comment:
Update this to give yourself credit Cosmin : D

Reply:
Please!